presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -475,7 +483,20 @@ public CounterStat getHighMemorySplitSource()
 
                 results.add(new HivePartitionMetadata(hivePartition, Optional.of(partition), partitionSchemaDifference.build(), encryptionInformation));
             }
-
+            String warningMessage = format("Table '%s' has '%s' out of '%s' partitions unreadable: ", tableName, unreadablePartitionsSkipped, partitionBatch.size());
+            for (Entry<String, Set<String>> entry : partitionsNotReadable.entrySet()) {
+                if (entry.getValue().size() > 2) {
+                    Iterator it = entry.getValue().iterator();
+                    warningMessage += "\n" + entry.getKey().toString() + ": " + it.next() + ", " + it.next() + " ...";
+                }
+                else {
+                    warningMessage += "\n" + entry.getKey().toString() + ": " + entry.getValue().toString();
+                }
+            }
+            if (partitionsNotReadable.entrySet().size() > 2) {
+                warningMessage += "\n...";
+            }
+            warningCollector.add(new PrestoWarning(PARTITION_NOT_READABLE, warningMessage));
```

Comment:
In test `testPartitionNotReadable`, right below `assertEquals(SPLIT_SCHEDULING_CONTEXT.getWarningCollector().getWarnings().size(), 1);`,  assert the error message popped out is the one you would expect.